binapigen/generator.go

```diff
@@ -171,6 +231,13 @@ type GenFile struct {
 	packageNames  map[GoImportPath]GoPackageName
 }
 
+func (g *GenFile) SetFile(f *File) { g.file = f }
```

Comment:
From the quick glance, it seems like we don't really need setter/getters here. This field is only used internally for generating the main API file. I do not see any reason for needing this exported. Any caller of `NewGenFile` knows what API file they based it on (if any) and can keep this reference locally.

Reply:
Agreed for the Setter, it is cleaner called from the constructor.
Although we need a Getter for the plugins that depend from the file content (e.g. gen_rpc does `FPrintf("defines RPC service ", g.GetFile().Desc.Name)` )